date-picker: Split `DateRangePicker` into own entrypoint (#1163)
steelthreads/agds-next: packages/react/src/date-picker/docs/overview.mdx

---
title: Date picker

> description: The Date picker component allows users to select a single date via a calendar or text input.

Reviewer: Do we have some kind of guidance to say `Date picker` instead of `DatePicker` when talking about the component? This always feels weird to me.

Author: The convention is to use sentence case (e.g. `Date picker`) and this recommendation came from the content team. 

This component documentation (and many others) was written before the recommendation though.

group: forms
storybookPath: /story/forms-datepicker-datepicker--basic
figmaGalleryNodeId: 12444%3A100327
relatedComponents: ['date-range-picker']
---

```jsx live
() => {
	const [value, setValue] = React.useState();
	return <DatePicker label="Select date" value={value} onChange={setValue} />;
};
```

Date picker is a [controlled component](https://reactjs.org/docs/forms.html#controlled-components) which means consumers of this component need to manage the state of this component by using the `value`, `onChange` and `onInputChange` props.

For an example of using this component in a form built with `react-hook-form` and `yup`, please see the [Single-page form template](/patterns/single-page-form).

<DoHeading />

- always display the date format above the text input
- use Australian date format 'dd/mm/yyyy'
- allow users to navigate dates via the calendar
- ensure users can enter dates via the calendar or text input
- Make sure error messages are specific about what input is required.

<DontHeading />

> - never hide the date format

Reviewer: double negative here "don't never"
```suggestion
- hide the date format
```

- use international date formats such as USA mm/dd/yyyy
- hijack built-in keyboard navigation behaviour.

## Tracking the input value

The Date picker component has two methods of input:

1. Typing in a date string (dd/mm/yyyy) via the text input
2. Selecting a date via the calendar widget

As we do not have any sort of input masking due to accessibility concerns, there is a possibilty for the user to enter an invalid value via the text input. In some scenarios, it can be useful to know what the value of the text input is so the user can be informed of any invalid values.

> For example, if the Date picker component was placed inside of a form an optional field. If the user typed in an invalid value via the text input, the `onChange` event would never fire. This may lead the user to believe what they have typed is valid when actually their input will not be saved.

Reviewer: ```suggestion
For example, if the Date picker component was an optional field. If a user inputs an invalid value via the text input, the `onChange` event will never fire. This may lead the user to believe what they have typed is valid when actually their input will not be saved.
```
